src/Mod/Draft/draftguitools/gui_selectplane.py

```diff
@@ -96,7 +96,10 @@ def Activated(self):
 
         # Fill values
         self.taskd.form.checkCenter.setChecked(self.param.GetBool("CenterPlaneOnView", False))
-        q = FreeCAD.Units.Quantity(self.param.GetFloat("gridSpacing", 1.0), FreeCAD.Units.Length)
+        try:
+            q = FreeCAD.Units.Quantity(self.param.GetString("gridSpacing", "1 mm"))
+        except ValueError:
+            q = FreeCAD.Units.Quantity("1 mm")
```

Comment:
I just saw that it's possible to have two values with the same name:
![Screenshot from 2022-05-19 13-10-09](https://user-images.githubusercontent.com/1136856/169280271-2cb56263-be24-4e76-971c-2f60987f0702.png)
What about this:
```suggestion
        try:
            q = FreeCAD.Units.Quantity(self.param.GetString("gridSpacing", "1 mm"))
        except ValueError:
            # older format, for backwards compatibility
            q = FreeCAD.Units.Quantity(self.param.GetFloat("gridSpacing", 1.0), FreeCAD.Units.Length)
```
Since the value is then saved as string, next time everything will be fine. But maybe it's not worth it, after all it's not terrible if the grid is reset to 1... 

This needs to be changed too: https://github.com/yorikvanhavre/BIM_Workbench/blob/master/BimSetup.py#L144 and L343

Reply:
> What about this:

No strong opinion on my side. :) I'm not really a Draft user. Generally, I don't like to have to maintain code written just for transitional things.